### DISCUSSION TITLE
AWS-CDK How to update DynamoDB capacity unit

### DISCUSSION THREAD

**Discussion by jerzystachera** [aws/aws-cdk#27161] [General]
I've tried to update DynamoDB capacity units but it had no effect:

`new Table(this, props.projectName,
  {
      tableName: props.tableName,
      partitionKey: {name: props.partitionKeyName, type: AttributeType.STRING},
      sortKey: {name: props.sortKeyName, type: AttributeType.STRING},
      removalPolicy: RemovalPolicy.DESTROY,
      stream: StreamViewType.NEW_AND_OLD_IMAGES,
      readCapacity: 15,
      writeCapacity: 15
  });
`
            
The table still has the default one values: 5 read and 5 write capacity units.

How can I solve this ?

**Comment by khushail** [comment]
@jerzystachera , as mentioned in the docs [here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_dynamodb.Table.html#billingmode) `billing mode` should be provisioned for changing the capacity. Could you check if the [billing mode is provisioned](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_dynamodb.Table.html#writecapacity) in your scenario.

**Reply by jerzystachera** [reply]
The billing mode is provisioned.
The problem is that I cannot update the DynamoDB auto-scaling options after the table was created.
Here is the full code with marked sections that cause the problem (the newly added)

```

// create table
this.dynamoDbTable = new Table(this, props.projectName,
	{
		tableName: props.tableName,
		partitionKey: {name: props.partitionKeyName, type: AttributeType.STRING},
		sortKey: {name: props.sortKeyName, type: AttributeType.STRING},
		removalPolicy: RemovalPolicy.DESTROY,
		stream: StreamViewType.NEW_AND_OLD_IMAGES,
		billingMode: BillingMode.PROVISIONED,
		readCapacity: dynamoDbMinCu,
		writeCapacity: dynamoDbMinCu
	});

// here is the problem <<<<<<<-----------------
this.dynamoDbTable.autoScaleReadCapacity(
	this.createDefaultScalingProps(dynamoDbMinCu, dynamoDbMaxCu)
);
this.dynamoDbTable.autoScaleWriteCapacity(
	this.createDefaultScalingProps(dynamoDbMinCu, dynamoDbMaxCu)
);
// --------------------------------------------

// create global secondary indices
for (let gsi of props.globalSecondaryIndices) {
	this.dynamoDbTable.addGlobalSecondaryIndex(
		this.createGSIIndexProps(
			gsi.name,
			gsi.pak,
			gsi.sk,
			dynamoDbMinCu)
	);

	// here is the problem <<<<<<<-----------------
	this.dynamoDbTable.autoScaleGlobalSecondaryIndexReadCapacity(
		gsi.name,
		this.createDefaultScalingProps(dynamoDbMinCu, dynamoDbMaxCu)
	);

	this.dynamoDbTable.autoScaleGlobalSecondaryIndexWriteCapacity(
		gsi.name,
		this.createDefaultScalingProps(dynamoDbMinCu, dynamoDbMaxCu)
	);
	// --------------------------------------------
}

```

And when I try to deploy the code with added auto-scaling I got series of error 'Already exists':

- - "Resource of type 'AWS::ApplicationAutoScaling::ScalableTarget' with identifier 'table/mip-table/index/gsiX|dynamodb:index:WriteCapacityUnits|dynamodb' already exists."
- "Resource of type 'AWS::ApplicationAutoScaling::ScalableTarget' with identifier 'table/mip-table/index/gsiX|dynamodb:index:ReadCapacityUnits|dynamodb' already exists."
- "Resource of type 'AWS::ApplicationAutoScaling::ScalableTarget' with identifier 'table/mip-table|dynamodb:table:WriteCapacityUnits|dynamodb' already exists."

What is the solution for that ? how can I update the dynamodb auto-scaling options ?
